de-lib/src/main/java/org/iplantc/de/admin/desktop/client/services/impl/AppAdminServiceFacadeImpl.java

```diff
@@ -250,4 +261,27 @@ protected AppDoc convertFrom(String object) {
         });
     }
 
+    @Override
+    public void searchApp(String term, AsyncCallback<AppListLoadResult> callback) {
+        StringBuilder address = new StringBuilder(APPS_ADMIN);
+        if(!Strings.isNullOrEmpty(term)) {
+            address.append("?search=" + URL.encodeQueryString(term));
+        }
+
+        ServiceCallWrapper wrapper = new ServiceCallWrapper(GET, address.toString());
+        deService.getServiceData(wrapper,  new AsyncCallbackConverter<String, AppListLoadResult>(callback) {
+            @Override
+            protected AppListLoadResult convertFrom(String object) {
+                List<App> apps = AutoBeanCodex.decode(svcFactory, AppList.class, object).as().getApps();
```

Comment:
I think you should be able to add `AppListLoadResult` to `AppServiceAutoBeanFactory` so you can do a direct conversion from String to `AppListLoadResult` instead of separating it into `List<Apps>` and then building `AppListLoadResult` manually.